### PR TITLE
Fix flaky test corehq.motech.tests.test_public_only_session

### PR DESCRIPTION
**Pull request by dannyroberts**
## Product Description
This is a test-only change

## Technical Summary
example.com was sometimes 404ing, which made this flaky.

### Safety story
This can't break functionality because only tests are changed, but it could theoretically make a test pass without testing what it's supposed to be. So I tested locally that the patch works the way I think it does by making it use a status code of something other than 200, and showing the test fails.

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by gherceg — APPROVED** [review]
Thanks Danny!

**Review by snopoke — COMMENTED** [review]
Doesn't this defeat the point of the tests entirely since it's bypassing the logic that's being tested.

File: corehq/motech/tests/test_public_only_session.py [hunk 1]
```diff
@@ -10,8 +10,10 @@
 
 def test_public_only_session__simple_success():
     session = _set_up_session()
-    response = session.get('https://example.com/')
-    eq(response.status_code, 200)
+    with _patch_session_with_hard_coded_response(session, 'https://example.com/',
+                                                 _get_200_response()):
+        response = session.get('https://example.com/')
+        eq(response.status_code, 200)
```
Comment (millerdev) [inline corehq/motech/tests/test_public_only_session.py:16]: Adding to @snopoke's question, what is this testing?

**Review by millerdev — APPROVED** [review]

File: corehq/motech/tests/test_public_only_session.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/motech/tests/test_public_only_session.py:16]: Okay I'll add that

**Pull request merged**